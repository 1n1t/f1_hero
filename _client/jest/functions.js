/* eslint-disable import/prefer-default-export */
export function sel(id) {
	return `[data-test="${id}"]`;
}
