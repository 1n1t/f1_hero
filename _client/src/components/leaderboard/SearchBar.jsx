import React from 'react';

const SearchBar = props => {
  return (
    <div className="search-bar">
      <input
        type="text"
        placeholder="Search..."
        onChange={props.onChange}
        value={props.searchTerm}
      />
      <i className="fa fa-search" aria-hidden="true"></i>
    </div>
  );
};

export default SearchBar;
