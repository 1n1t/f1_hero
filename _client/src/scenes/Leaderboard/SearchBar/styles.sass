.SearchBar
  margin-bottom: 30px
  position: relative

  input
    background-color: transparent
    border: none
    border-bottom: 1px solid white
    outline: none

    &, &::-webkit-input-placeholder
      color: white

    &::-webkit-input-placeholder
      font-weight: 300

    display: block
    width: calc(100% - 30px)
    text-align: center
    text-transform: lowercase
    font-size: 1rem
    padding-bottom: 10px
    padding-right: 30px

  .fa
    position: absolute
    right: 5px
    top: 0
