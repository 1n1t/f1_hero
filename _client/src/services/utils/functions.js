/* eslint-disable import/prefer-default-export */
export const isObjectEmpty = obj =>
	Object.keys(obj).length === 0 && obj.constructor === Object;

export const getNodePaddings = node => {
	const nodeStyle = window.getComputedStyle(node);
	return (
		parseInt(nodeStyle.getPropertyValue('padding-right'), 10) +
		parseInt(nodeStyle.getPropertyValue('padding-left'), 10)
	);
};
