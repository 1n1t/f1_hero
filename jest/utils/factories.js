/* eslint-disable no-use-before-define */
const mongoose = require('mongoose');
// eslint-disable-next-line import/no-extraneous-dependencies
const faker = require('faker');

const { capitalizeFirstLetter } = require('../../utils/functions');

const self = {
	createRace,
	createUser,
	createDriver,
	createPick
};

function create(model, fields = {}) {
	const fn = self[`create${capitalizeFirstLetter(model)}`];
	if (typeof fn === 'function') {
		return fn(fields);
	}
	throw new FactoryException(model);
}

function createRace(fields) {
	const Race = mongoose.model('race');

	return Race.create({
		round: faker.random.number(20),
		raceName: faker.random.words(3),
		country: faker.address.country(),
		date: faker.date.future(),
		alpha3code: `_${faker.address.countryCode()}`,
		flagUrl: faker.image.imageUrl(),
		...fields
	});
}

function createUser(fields) {
	const User = mongoose.model('user');

	return User.create({
		twitterID: faker.random.uuid(),
		username: faker.internet.userName(),
		...fields
	});
}

function createDriver(fields) {
	const Driver = mongoose.model('driver');
	const name = `${faker.name.firstName()} ${faker.name.lastName()}`;

	return Driver.create({
		number: faker.random.number(99),
		name,
		code: name.slice(0, 3).toUpperCase(),
		...fields
	});
}

async function createPick(fields) {
	const Pick = mongoose.model('pick');

	return Pick.create({
		_user: fields._user || (await self.createUser()),
		_race: fields._race || (await self.createRace()),
		forecast: [],
		...fields
	});
}

function FactoryException(model) {
	this.message = `A factory wasn't registered for the model "${model}"`;
}

const factory = {
	create
};

module.exports = factory;
