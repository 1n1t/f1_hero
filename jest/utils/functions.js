const mongoose = require('mongoose');

function clearCollection(model) {
	return mongoose.model(model).remove();
}

function clearAllCollections() {
	Promise.all([
		mongoose.model('race').remove(),
		mongoose.model('user').remove(),
		mongoose.model('pick').remove(),
		mongoose.model('driver').remove()
	]).then(() => null);
}

function sel(id) {
	return `[data-test="${id}"]`;
}

module.exports = {
	clearCollection,
	clearAllCollections,
	sel
};
