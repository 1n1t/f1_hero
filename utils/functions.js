function getMaxFiveElements(arr) {
	return arr.sort((a, b) => a < b).slice(0, 5);
}

function capitalizeFirstLetter(string) {
	return string.charAt(0).toUpperCase() + string.slice(1);
}

module.exports = {
	getMaxFiveElements,
	capitalizeFirstLetter
};
