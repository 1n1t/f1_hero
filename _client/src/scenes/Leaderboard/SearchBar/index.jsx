import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const SearchBar = ({ searchTerm, onChange }) => (
	<div className="SearchBar">
		<input
			type="text"
			placeholder="Search..."
			onChange={onChange}
			value={searchTerm}
			data-test="searchInput"
		/>
		<i className="fa fa-search" aria-hidden="true" />
	</div>
);

SearchBar.propTypes = {
	searchTerm: PropTypes.string.isRequired,
	onChange: PropTypes.func.isRequired
};

export default SearchBar;
