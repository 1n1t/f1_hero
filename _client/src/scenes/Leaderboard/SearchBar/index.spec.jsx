import r from 'Root/jest/utils/renderer';
import { sel } from 'Root/jest/utils/functions';
import SearchBar from './index';

const defaultProps = {
	searchTerm: '',
	onChange() {}
};

r.init(SearchBar, defaultProps);

it('renders correctly', () => {
	const wrapper = r.render(null, true);
	expect(wrapper).toMatchSnapshot();
});

it('displays passed search term', () => {
	const searchTerm = 'foo';
	const wrapper = r.render({ searchTerm }, true);
	const { value } = wrapper.find(sel('searchInput')).props();

	expect(value).toBe(searchTerm);
});

it('calls provided callback when the value is changed', () => {
	const onChange = jest.fn();
	const wrapper = r.render({ onChange }, true);
	const input = wrapper.find(sel('searchInput'));

	input.simulate('change');
	expect(onChange).toHaveBeenCalledTimes(1);
});
