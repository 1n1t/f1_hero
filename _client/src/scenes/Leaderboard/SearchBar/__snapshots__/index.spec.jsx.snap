// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="SearchBar"
>
  <input
    data-test="searchInput"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
  <i
    aria-hidden="true"
    className="fa fa-search"
  />
</div>
`;
